pyroute2/netlink/__init__.py

```diff
@@ -1678,6 +1678,79 @@ class mpls_target(target):
 
         family = AF_MPLS
 
+    class seg6_target(target):
+
+        __slots__ = ()
+
+        SEG6_IPTUN_MODE_INLINE = 0
+        SEG6_IPTUN_MODE_ENCAP = 1
+
+        encapmodes = {
+          "inline": SEG6_IPTUN_MODE_INLINE,
+          "encap": SEG6_IPTUN_MODE_ENCAP
+        }
+
+        SEG6_TYPE = 4
+
+        def encode(self):
+
+            # Retrieve the family
+            family = self.get_family()
+            # Seg6 can be applied only to IPv6
+            if family == AF_INET6:
+
+                # TODO other seg6 functionalities:
+                # Inline mode
+                # flags
+                # hmac
+                # ...
+
+                # Get the encap
+                encap = self.value[0]
+                # Get mode
+                mode = encap.get('mode', None)
+                # Get segs
+                segs = encap.get('segs', [])
+                # Initialize value
+                self['value'] = b''
+                # Add mode to value
+                self['value'] += struct.pack('i', self.encapmodes.get(mode, 1))
+                # Nexthdr is not set
+                nexthdr = 0
+                # Add nexthdr to value
+                self['value'] += struct.pack('B', nexthdr)
+                # Calculate srlen
+                srhlen = 8 + 16 * len(segs);
+                # Calculate hdrlen
+                hdrlen = (srhlen >> 3) - 1;
+                # Add hdrlen to value
+                self['value'] += struct.pack('B', hdrlen)
+                # Add type to value
+                self['value'] += struct.pack('B', self.SEG6_TYPE)
+                # Calculate segments left
+                segments_left = len(segs) - 1;
```

Comment:
what if we have 10 segments?

Reply:
I didn't get. Why 10 should be a problem ? 

Btw, ";" is not necessary

Reply:
Just trying to understand, to which segment does it point.

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/shemminger/iproute2.git/tree/ip/iproute_lwtunnel.c#n467

Reply:
Ok, thanks!